adhocracy4/follows/static/follow_dropdown/FollowDropdown.jsx

```diff
@@ -0,0 +1,87 @@
+var api = require('../../../static/api')
+var django = require('django')
+var React = require('react')
+
+class FollowDropdown extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      followed: undefined,
+      follows: 0
+    }
+  }
+  setFollow (enabled) {
+    if (enabled === this.state.followed) {
+      return false
+    }
+
+    api.follow.change({ enabled: enabled }, this.props.project)
+      .done((follow) => {
+        this.setState({
+          followed: follow.enabled,
+          follows: follow.follows
+        })
+      })
+  }
+  enableFollow () {
+    this.setFollow(true)
+  }
+  disableFollow () {
+    this.setFollow(false)
+  }
+  componentDidMount () {
+    api.follow.get(this.props.project)
+      .done((follow) => {
+        this.setState({
+          followed: follow.enabled,
+          follows: follow.follows
+        })
+      })
+      .fail((response) => {
+        if (response.status === 404) {
+          this.setState({
+            followed: false
+          })
+        }
+      })
+  }
+  render () {
+    // FIXME: string extraction seems to be broken with JSX
+    var notFollowingLabel = django.gettext('Not following')
+    var notFollowingHelp = django.gettext('Never be notified.')
+    var followingLabel = django.gettext('Following')
+    var followingHelp = django.gettext('Receive email notifications about this project.')
```

Comment:
He this is funny :) this is actually the solution to the translation problems from yesterday

Reply:
Ahh just adding the strings as variables?

Reply:
Yeah...I'm still checking how other projects handle this though :/
The good thing about this solution is, apart from being tiresome, it's quite future proof and doesn't rely on an ever more complex npm setup. So maybe we should start writing our jsx files like this by default.

Reply:
cunning! and good to know i'll add the button wording like that in the new follow component